src/scf/self_consistent_field.jl

```diff
@@ -1,7 +1,9 @@
 """
 Obtain new density ρ by diagonalizing `ham`.
 """
-function next_density(ham::Hamiltonian, n_bands; ψ=nothing,
+function next_density(ham::Hamiltonian;
+                      n_bands=div(ham.basis.model.n_electrons, filled_occupation(ham.basis.model)),
```

Comment:
I'm actually really annoyed by all these nbands. Would it be a bad idea if we put it in `model` or (probably better) in `basis`?

Reply:
- This only turns up in two places. It's only everywhere right now, since we are missing a sensible heuristic for the metallic case. Once we invent that, we are good and they disappear from all examples as well (given such a  `default_n_bands(model)` function).
- Re bad idea putting it in the `model`: I think it is, because it's not really part of the model or the basis. I mean in principle you might even want to adapt it during the SCF to reflect the changes to the occupation and save on computing "unnecessary" bands. Or if I am interested in more bands, I might turn it up in a second SCF run or so ... maybe weird, but we should allow for that, I think.

Reply:
Good point for the second point. Agreed we need a good heuristic in the metallic case (or a good way to increase/decrease it on the fly).